sale_order_line_position/models/position_mixin.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class OrderPositionMixin(models.AbstractModel):
+    _name = "order.position.mixin"
+    _description = "Order position mixin"
+
+    locked_positions = fields.Boolean(compute="_compute_locked_positions")
+
+    @api.depends("state")
+    def _compute_locked_positions(self):
+        for record in self:
+            record.locked_positions = record.state != "draft"
+
+    def action_confirm(self):
+        self.recompute_positions()
+        return super().action_confirm()
+
+    def action_quotation_send(self):
+        self.recompute_positions()
+        return super().action_quotation_send()
+
+    def recompute_positions(self):
+        return NotImplementedError
```

Comment:
```suggestion
    def recompute_positions(self):
        if not self.locked_positions:
            self._recompute_positions()
    
    def _recompute_positions(self):
        return NotImplementedError
```

See https://github.com/OCA/sale-reporting/pull/195